Failure of :modules:transport-netty4:test on master

Build | 20191022054831-A81DC5D0
Log | https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/os=windows-2012-r2/215/console
Build Scans | https://gradle-enterprise.elastic.co/s/hpnzc7hnzftce

Reproduce line:
```
REPRODUCE WITH: gradlew ':modules:transport-netty4:test' --tests "org.elasticsearch.transport.netty4.SimpleNetty4TransportTests.testLocalNodeConnection" \
  -Dtests.seed=175DFFF16E9928FC \
  -Dtests.security.manager=true \
  -Dtests.locale=lg-UG \
  -Dtests.timezone=Africa/Juba \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

Does *not* reproduce locally.


Excerpt from logs:
```
06:39:55   2> java.lang.AssertionError: expected:<0> but was:<16777216>
06:39:55         at __randomizedtesting.SeedInfo.seed([175DFFF16E9928FC:6655EB5329A51DB6]:0)
06:39:55         at org.junit.Assert.fail(Assert.java:88)
06:39:55         at org.junit.Assert.failNotEquals(Assert.java:834)
06:39:55         at org.junit.Assert.assertEquals(Assert.java:645)
06:39:55         at org.junit.Assert.assertEquals(Assert.java:631)
06:39:55         at org.elasticsearch.transport.netty4.SimpleNetty4TransportTests.tearDown(SimpleNetty4TransportTests.java:53)

```

--- a/modules/transport-netty4/src/test/java/org/elasticsearch/ESNetty4IntegTestCase.java
+++ b/modules/transport-netty4/src/test/java/org/elasticsearch/ESNetty4IntegTestCase.java
@@ -18,7 +18,6 @@
  */
 package org.elasticsearch;
 
-import io.netty.buffer.PooledByteBufAllocator;
 import org.elasticsearch.common.network.NetworkModule;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.plugins.Plugin;
@@ -56,12 +55,5 @@ public abstract class ESNetty4IntegTestCase extends ESIntegTestCase {
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
         return Collections.singletonList(Netty4Plugin.class);
-    }
-
-    @Override
-    public void tearDown() throws Exception {
-        super.tearDown();
-        assertEquals(0, PooledByteBufAllocator.DEFAULT.metric().usedHeapMemory());
-        assertEquals(0, PooledByteBufAllocator.DEFAULT.metric().usedDirectMemory());
     }
 }

--- a/modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java
+++ b/modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java
@@ -25,6 +25,7 @@ import io.netty.buffer.Unpooled;
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelHandlerAdapter;
 import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
 import io.netty.channel.nio.NioEventLoopGroup;
 import io.netty.channel.socket.SocketChannel;
 import io.netty.channel.socket.nio.NioSocketChannel;
@@ -61,6 +62,7 @@ import org.elasticsearch.rest.RestRequest;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.threadpool.TestThreadPool;
 import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.NettyAllocator;
 import org.junit.After;
 import org.junit.Before;
 
@@ -348,7 +350,10 @@ public class Netty4HttpServerTransportTests extends ESTestCase {
 
             CountDownLatch channelClosedLatch = new CountDownLatch(1);
 
-            Bootstrap clientBootstrap = new Bootstrap().channel(NioSocketChannel.class).handler(new ChannelInitializer<SocketChannel>() {
+            Bootstrap clientBootstrap = new Bootstrap()
+                .option(ChannelOption.ALLOCATOR, NettyAllocator.getAllocator())
+                .channel(NioSocketChannel.class)
+                .handler(new ChannelInitializer<SocketChannel>() {
 
                 @Override
                 protected void initChannel(SocketChannel ch) {

--- a/modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/SimpleNetty4TransportTests.java
+++ b/modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/SimpleNetty4TransportTests.java
@@ -19,7 +19,6 @@
 
 package org.elasticsearch.transport.netty4;
 
-import io.netty.buffer.PooledByteBufAllocator;
 import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.cluster.node.DiscoveryNode;
@@ -45,13 +44,6 @@ import static java.util.Collections.emptySet;
 import static org.hamcrest.Matchers.containsString;
 
 public class SimpleNetty4TransportTests extends AbstractSimpleTransportTestCase {
-
-    @Override
-    public void tearDown() throws Exception {
-        super.tearDown();
-        assertEquals(0, PooledByteBufAllocator.DEFAULT.metric().usedHeapMemory());
-        assertEquals(0, PooledByteBufAllocator.DEFAULT.metric().usedDirectMemory());
-    }
 
     @Override
     protected Transport build(Settings settings, final Version version, ClusterSettings clusterSettings, boolean doHandshake) {

